pkg/server/testserver.go

```diff
@@ -474,7 +477,10 @@ func (ts *testServer) SQLConnE(opts ...serverutils.SQLConnOption) (*gosql.DB, er
 		catconstants.SystemTenantName,
 		ts.Stopper(),
 		ts.topLevelServer.loopbackPgL,
-		ts.cfg.SQLAdvertiseAddr,
+		// Use the SQLAddr here. The SQLAddr is the literal address that the server
```

Comment:
hrm. I'm not an expert in what is expected, but it seems odd to _not_ use the advertised address when opening sql connections. I understand why you need this for your specific test, but this change seems like it could have a large blast radius. As a different approach, could you use a test hook to replace the partition addresses generated by the source cluster with the black hole addr you need for the test?

Reply:
You are right that this makes more sense as a test hook. I added a hook that mutates the node information returned by the stream producer and rolled back the change to the test server.